presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/TypeUtils.java

```diff
@@ -0,0 +1,261 @@
+import static io.prestosql.spi.type.TimeType.TIME;
+import static io.prestosql.spi.type.TimeWithTimeZoneType.TIME_WITH_TIME_ZONE;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+import static io.prestosql.spi.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+import static io.prestosql.spi.type.TinyintType.TINYINT;
+import static io.prestosql.spi.type.VarbinaryType.VARBINARY;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.prestosql.spi.type.VarcharType.createVarcharType;
+import static io.prestosql.spi.type.Varchars.isVarcharType;
+import static java.lang.Float.intBitsToFloat;
+import static java.lang.Math.min;
+import static java.lang.Math.toIntExact;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static org.joda.time.chrono.ISOChronology.getInstanceUTC;
+
+public final class TypeUtils
+{
+    private static final Map<Type, String> SQL_TYPES = ImmutableMap.<Type, String>builder()
+            .put(BOOLEAN, "BOOLEAN")
+            .put(BIGINT, "BIGINT")
+            .put(INTEGER, "INTEGER")
+            .put(SMALLINT, "SMALLINT")
+            .put(TINYINT, "TINYINT")
+            .put(DOUBLE, "DOUBLE")
+            .put(REAL, "FLOAT")
+            .put(VARBINARY, "VARBINARY")
+            .put(DATE, "DATE")
+            .put(TIME, "TIME")
+            .put(TIME_WITH_TIME_ZONE, "TIME")
+            .put(TIMESTAMP, "TIMESTAMP")
+            .put(TIMESTAMP_WITH_TIME_ZONE, "TIMESTAMP")
+            .build();
+
+    private TypeUtils()
+    {
+    }
+
+    public static boolean isArrayType(Type type)
+    {
+        return type.getTypeSignature().getBase().equals(StandardTypes.ARRAY);
+    }
+
+    public static boolean isMapType(Type type)
+    {
+        return type.getTypeSignature().getBase().equals(StandardTypes.MAP);
+    }
+
+    public static boolean isRowType(Type type)
+    {
+        return type.getTypeSignature().getBase().equals(StandardTypes.ROW);
+    }
+
+    public static boolean isDateType(Type type)
+    {
+        return type.equals(DATE) ||
+                type.equals(TIME) ||
+                type.equals(TIMESTAMP) ||
+                type.equals(TIMESTAMP_WITH_TIME_ZONE);
+    }
+
+    public static String toSqlType(Type type)
+    {
+        if (type instanceof VarcharType) {
+            if (((VarcharType) type).isUnbounded()) {
+                return "VARCHAR";
+            }
+            return "VARCHAR(" + ((VarcharType) type).getBoundedLength() + ")";
+        }
+        if (type instanceof CharType) {
+            if (((CharType) type).getLength() == CharType.MAX_LENGTH) {
+                return "CHAR";
+            }
+            return "CHAR(" + ((CharType) type).getLength() + ")";
+        }
+
+        String sqlType = SQL_TYPES.get(type);
+        if (sqlType != null) {
+            return sqlType;
+        }
+
+        if (type instanceof DecimalType) {
+            return type.toString().toUpperCase();
```

Comment:
Don't use toString() 

```suggestion
            return format("DECIMAL(%s, %s)", ....
```